streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -64,6 +66,7 @@ public ClientState() {
         previousStandbyTasks.taskIds(new TreeSet<>());
         previousActiveTasks.taskIds(new TreeSet<>());
 
+        clientTags = new HashMap<>();
```

Comment:
I guess it would be better to move the assignment to the field declaration to avoid duplication among constructors.

Reply:
Good point about client tags being constant. Added it as constructor parameter.